cordova-lib/src/plugman/fetch.js

```diff
@@ -70,6 +70,10 @@ function fetchPlugin(plugin_src, plugins_dir, options) {
             //if --fetch was used, throw error for subdirectories
             if (result[2] && options.fetch) {
                 return Q.reject(new CordovaError('--fetch does not support subdirectories'));
+            } else {
+                if(result[2]) {
+                    events.emit('warn', 'support for subdirectories is depreated and will be removed in Cordova@7');
+                }
```

Comment:
if options.subdir *and* options.fetch are both set, this warning will never be printed. This warning should be emitted in the code block here instead: https://github.com/sterlingann/cordova-lib/blob/497a6d57f5c4ebb324ce6a838dd96a5881a21a75/cordova-lib/src/plugman/fetch.js#L69